runelite-client/src/main/java/net/runelite/client/util/LinkBrowser.java

```diff
@@ -48,37 +49,67 @@
 	/**
 	 * Tries to navigate to specified URL in browser. In case operation fails, displays message box with message
 	 * and copies link to clipboard to navigate to.
-	 * @param url url to open
-	 * @return true if operation was successful
 	 */
-	public static boolean browse(final String url)
+	public static void browse(final String url)
 	{
-		if (Strings.isNullOrEmpty(url))
+		new Thread(() -> 
 		{
-			return false;
-		}
+			if (Strings.isNullOrEmpty(url))
+			{
+				log.warn("LinkBrowser.browse() called with invalid input");
+				return;
+			}
 
-		if (attemptDesktopBrowse(url))
-		{
-			log.debug("Opened browser through Desktop#browse to {}", url);
-			return true;
-		}
+			if (attemptDesktopBrowse(url))
+			{
+				log.debug("Opened directory through Desktop#browse to {}", url);
+				return;
+			}
+
+			if (shouldAttemptXdg && attemptXdgOpen(url))
+			{
+				log.debug("Opened directory through xdg-open to {}", url);
+				return;
+			}
+			log.warn("LinkBrowser.browse() could not open {}", url);
+			showMessageBox("Unable to open link. Press 'OK' and the link will be copied to your clipboard.", url);
+		}).start();
+	}
 
-		if (shouldAttemptXdg && attemptXdgOpen(url))
+	/**
+	 * Tries to open a directory in the OS native file manager.
+	 * @param directory directory to open
+	 */
+	public static void open(final String directory)
+	{
+		new Thread(() ->
 		{
-			log.debug("Opened browser through xdg-open to {}", url);
-			return true;
-		}
+			if (Strings.isNullOrEmpty(directory))
+			{
+				log.warn("LinkBrowser.open() called with invalid input");
+				return;
+			}
+
+			if (attemptDesktopOpen(directory))
+			{
+				log.debug("Opened directory through Desktop#open to {}", directory);
+				return;
+			}
 
-		showMessageBox("Unable to open link. Press 'OK' and link will be copied to your clipboard.", url);
-		return false;
+			if (shouldAttemptXdg && attemptXdgOpen(directory))
+			{
+				log.debug("Opened directory through xdg-open to {}", directory);
+				return;
+			}
+			log.warn("LinkBrowser.open() could not open {}", directory);
```

Comment:
Maybe this should show messagebox as well to let the user know, and maybe copy target path to clipboard too

Reply:
Add newline above